pkg/optjson/stringor.go

```diff
@@ -0,0 +1,31 @@
+package optjson
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// StringOr is a JSON value that can be a string or a different type of object
+// (e.g. somewhat common for a string or an array of strings, but can also be
+// a string or an object, etc.).
+type StringOr[T any] struct {
```

Comment:
So this is not technically an "optional JSON" type, but I feel it fits better in this package than floating elsewhere or creating another JSON-related package. Maybe we should rename the package, but I didn't go that far. Happy to adjust if anyone has a strong preference!